src/EnergyPlus/WindowManager.cc

```diff
@@ -768,25 +790,103 @@ namespace WindowManager {
+						ShowWarningError( "Window glazing material \"" + Material( LayPtr ).Name + "\" was defined with full spectral and angular data and has been converted to average spectral data" );
+						ShowContinueError( "due to its use with between-glass shades or blinds of the window construction \"" + Construct( ConstrNum ).Name + "\"." );
+						ShowContinueError( "All occurrences of this glazing material will be modeled as SpectralAverage." );
+						ShowContinueError( "If this material is also used in other window constructions  without between-glass shades or blinds," );
+						ShowContinueError( "then make a duplicate material (with new name) if you want to model those windows  (and reference the new material) using the full spectral data." );
+						// calc Trans, TransVis, ReflectSolBeamFront, ReflectSolBeamBack, ReflectVisBeamFront, ReflectVisBeamBack
+						//  assuming wlt same as wle
+						int NumX2 = TableLookup( PerfCurve( Material( LayPtr ).GlassSpecAngTransDataPtr ).TableIndex ).NumX2Vars;
+						for ( ILam = 1; ILam <= NumX2; ++ILam ) {
+							wlt( IGlass, ILam ) = TableData( PerfCurve( Material( LayPtr ).GlassSpecAngTransDataPtr ).TableIndex ).X2( ILam );
+							t( IGlass, ILam ) = TableData( PerfCurve( Material( LayPtr ).GlassSpecAngTransDataPtr ).TableIndex ).Y( ILam );
+							rff( IGlass, ILam ) = TableData( PerfCurve( Material( LayPtr ).GlassSpecAngFRefleDataPtr ).TableIndex ).Y( ILam );
+							rbb( IGlass, ILam ) = TableData( PerfCurve( Material( LayPtr ).GlassSpecAngBRefleDataPtr ).TableIndex ).Y( ILam );
+						}						
+						SolarSprectrumAverage( t, tmpTrans );
+						SolarSprectrumAverage( rff, tmpReflectSolBeamFront );
+						SolarSprectrumAverage( rbb, tmpReflectSolBeamBack );
+
+						// visible properties
+						VisibleSprectrumAverage( t, tmpTransVis );
+						VisibleSprectrumAverage( rff, tmpReflectVisBeamFront );
+						VisibleSprectrumAverage( rbb, tmpReflectVisBeamBack );
+
+						// set this material to average spectral data
+						Material( LayPtr ).GlassSpectralAndAngle = false;
+						Material( LayPtr ).Trans = tmpTrans;
+						Material( LayPtr ).TransVis = tmpTransVis;
+						Material( LayPtr ).ReflectSolBeamFront = tmpReflectSolBeamFront;
+						Material( LayPtr ).ReflectSolBeamBack = tmpReflectSolBeamBack;
+						Material( LayPtr ).ReflectVisBeamFront = tmpReflectVisBeamFront;
+						Material( LayPtr ).ReflectVisBeamBack = tmpReflectVisBeamBack;
+						SpecDataNum = 0;
+					}
+				}
 			} // End of loop over glass layers in the construction for front calculation
 
+			// Perform do loop to find common incident angles if multiple SpectralAngular layers are specified
+			for ( IGlass = 1; IGlass <= NGlass; ++IGlass ) {
+				LayPtr = Construct( ConstrNum ).LayerPoint( LayerNum( IGlass ) );
+				if ( Material( LayPtr ).GlassSpectralAndAngle ) {
+					AngleNum( IGlass ) = LayerNum( IGlass );
+				}
+			}
+			SetCommonIncidentAngles( ConstrNum, NGlass, TotalIPhi, AngleNum );
```

Comment:
@mjwitte Actually, this check needs to be performed as long as a single layer is SpectralAndAngle. For example, one layer is Spectral and other layer is SpectralAndAngle. If none of layers is SpectralAndAngle, the function will return without setting common incident angle. This is checked inside the function.